app/routes/goal.py

```diff
@@ -0,0 +1,109 @@
+from flask import Blueprint, jsonify, request
+from app import db
+
+from app.models.goal import Goal
+from app.models.task import Task
+from app.routes.routes_helper import get_valid_item_by_id
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goals_bp.route("", methods=['POST'])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except KeyError:
+        return {
+            "details": "Invalid data"
+            }, 400
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+
+@goals_bp.route("", methods=['GET'])
+def handle_goals():
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        goals = Goal.query.filter_by(title=title_query)
+    elif sort_query == "asc":
+        goals = Goal.query.order_by(Goal.title)
+    elif sort_query == "desc":
+        goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+
+    goals_response = [goal.to_dict() for goal in goals]
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=['GET'])
+def handle_goal(goal_id):
+    goal = get_valid_item_by_id(Goal, goal_id)
+    return {"goal": goal.to_dict()}, 200
+
+
+@goals_bp.route("/<goal_id>", methods=['PUT'])
+def update_one_goal(goal_id):
+    request_body = request.get_json()
+
+    goal_to_update = get_valid_item_by_id(Goal, goal_id)
+    goal_to_update.title = request_body["title"]
+
+    db.session.commit()
+
+    return {"goal": goal_to_update.to_dict()}, 200
+
+
+@goals_bp.route("/<goal_id>", methods=['DELETE'])
+def delete_one_goal(goal_id):
+    goal_to_delete = get_valid_item_by_id(Goal, goal_id)
+
+    db.session.delete(goal_to_delete)
+    db.session.commit()
+
+    return {
+        "details":
+            f'Goal {goal_to_delete.goal_id} "{goal_to_delete.title}" successfully deleted'
+        }, 200
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=['POST'])
+def post_task_ids_to_goal(goal_id):
+    goal = get_valid_item_by_id(Goal, goal_id)
+
+    request_body = request.get_json()
+    # Get the task ids from the request body
+    task_ids = request_body["task_ids"]
+
+    # Get the Task instances (task dicts) based on the task ids
+    task_instances = [get_valid_item_by_id(Task, task_id) for task_id in task_ids]
+    goal.tasks.extend(task_instances)
+
+    db.session.commit()
+
+    return {
+        "id": goal.to_dict()["id"],
+        "task_ids": task_ids
+    }, 200
```

Comment:
Your code style is great; your list comprehensions and comments only bring clarity and not too much noise or confusion, `extend` has a great usecase here, and your inline `return` statement is formatted cleanly! I'd be a very happy developer if we worked on the same codebase.